rest/data/generate.go

```diff
@@ -13,29 +14,29 @@ import (
 type GenerateConnector struct{}
 
 // GenerateTasks parses JSON files for `generate.tasks` and creates the new builds and tasks.
-func (gc *GenerateConnector) GenerateTasks(taskID string, jsonBytes []json.RawMessage) error {
+func (gc *GenerateConnector) GenerateTasks(taskID string, jsonBytes []json.RawMessage) (int, error) {
```

Comment:
Why not just return rest.APIError's from this function

Reply:
I like that idea better than returning HTTP codes from the model/data access layer. 

If you need more than on error I'd use common error values with error.Cause()